tests/compile-fail/concurrency/libc_pthread_join_multiple.rs

```diff
@@ -0,0 +1,30 @@
+// ignore-windows: Concurrency on Windows is not supported yet.
+
+// Joining the same thread multiple times is undefined behavior.
```

Comment:
I do not understand the difference between this and "libc_pthread_join_joined".

Reply:
`libc_pthread_join_joined.rs` the main thread joins the same child twice; `libc_pthread_join_multiple.rs` two different threads try to join the same thread.

Reply:
Ah I see, that makes sense.

This sounds very similar, *except* if in the "two different threads" case we can make it so that both threads do this concurrently, i.e, both start joining before the joined-on thread terminates. Can we make sure that is the case?

Reply:
> Can we make sure that is the case?

I have changed  that it is guaranteed to be the case with the current scheduler and likely to be the case with a random scheduler.